orchagent/portsorch.cpp

```diff
@@ -1002,6 +1034,13 @@ bool PortsOrch::setPortAdminStatus(Port &port, bool state)
     attr.id = SAI_PORT_ATTR_ADMIN_STATE;
     attr.value.booldata = state;
 
+    /* Update the host_tx_ready to false before setting admin_state, when admin state is false */
+    if (!state) 
+    {
+        m_portStateTable.hset(port.m_alias, "host_tx_ready", "false");
+        SWSS_LOG_INFO("Set admin status false %s to port pid:%" PRIx64, state ? "true" : "false", port.m_port_id);
+    }
+    
     sai_status_t status = sai_port_api->set_port_attribute(port.m_port_id, &attr);
     if (status != SAI_STATUS_SUCCESS)
```

Comment:
we are not handling this failure case, i think this is where the ASIC honors the "admin" status.

Reply:
This handling is existing code, can you please explain the comment?

Reply:
host_tx_ready is set to false when there is a SAI attribute failure.